lib/messagebird/client.rb

```diff
@@ -30,10 +30,12 @@ def initialize(errors)
   class Client
     attr_reader :access_key, :http_client, :conversation_client
 
-    def initialize(access_key = nil, http_client = nil)
+    ENABLE_CONVERSATIONS_WHATSAPP_SANDBOX = "ENABLE_CONVERSATIONS_WHATSAPP_SANDBOX"
+
+    def initialize(access_key = nil, http_client = nil, features = [])
       @access_key = access_key || ENV['MESSAGEBIRD_ACCESS_KEY']
       @http_client = http_client || HttpClient.new(@access_key)
-      @conversation_client = http_client || ConversationClient.new(@access_key)
+      @conversation_client = http_client || ConversationClient.new(@access_key, features)
```

Comment:
If consumers provide their own `http_client`, the `features` are ignored. I guess this can lead to confusing behaviour: can we somehow change the implementation to prevent this?

Reply:
The normal conversations API won't work either if the consumer provide their own `http_client`.
We can make a change that the http_client receives the base URL with every request, but that will also break backwards compatibility.

I'm open to suggestions on how to change this.

Reply:
Couple of options that come to mind:

1. Make the http_client receive the base URL, but default to REST API when it is not provided
2. Change the signature for client to `def initialize(access_key = nil, http_client = nil, conversation_client = nil)`, and do the same trick we currently do for http_client with the conversation_client. Let consumers pass the sandbox feature flag to `conversation_client` themselves.

I think (2) makes more sense - what do you think?

Also, this should've been caught by tests. Perhaps we should add those.

Reply:
Yes, I think 2 makes more sense. I'll change the constructor and then add a method to allow for enabling flags/features.

It looks like the main reason why they constructor allows for a http_client to be provided is for the tests to mock their own http_client, which is also the reason why the test doesn't pick this up.

I'll look at how to add this to the tests.